cognite/neat/data_model/models/entities/_single_value.py

```diff
@@ -0,0 +1,150 @@
+from pydantic import (
+    BaseModel,
+    field_validator,
+    model_serializer,
+)
+
+from cognite.neat.data_model._constants import (
+    EntityTypes,
+)
+
+from ._constants import (
+    Undefined,
+    Unknown,
+    _UndefinedType,
+    _UnknownType,
+)
+
+
+@total_ordering
+class ConceptualEntity(BaseModel, extra="ignore"):
+    """Conceptual Entity is a concept, class or property in semantics sense."""
+
+    type_: ClassVar[EntityTypes] = EntityTypes.undefined
+    prefix: str | _UndefinedType = Undefined
+    suffix: str
+
+    @model_serializer(when_used="unless-none", return_type=str)
+    def as_str(self) -> str:
+        return str(self)
+
+    @field_validator("*", mode="before")
+    def strip_string(cls, value: Any) -> Any:
+        if isinstance(value, str):
+            return value.strip()
+        elif isinstance(value, list):
+            return [entry.strip() if isinstance(entry, str) else entry for entry in value]
+        return value
+
+    def dump(self, **defaults: Any) -> str:
+        return self._as_str(**defaults)
+
+    def as_tuple(self) -> tuple[str, ...]:
+        # We haver overwritten the serialization to str, so we need to do it manually
+        extra: tuple[str, ...] = tuple(
+            [
+                str(v or "")
+                for field_name in self.model_fields.keys()
+                if (v := getattr(self, field_name)) and field_name not in {"prefix", "suffix"}
+            ]
+        )
+        if isinstance(self.prefix, _UndefinedType):
+            return str(self.suffix), *extra
+        else:
+            return self.prefix, str(self.suffix), *extra
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, ConceptualEntity):
+            return NotImplemented
+        return self.as_tuple() < other.as_tuple()
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, ConceptualEntity):
+            return NotImplemented
+        return self.as_tuple() == other.as_tuple()
+
+    def __hash__(self) -> int:
+        return hash(str(self))
+
+    def __str__(self) -> str:
+        return self.id
+
+    def __repr__(self) -> str:
+        # We have overwritten the serialization to str, so we need to do it manually
+        model_dump = ((k, v) for k in self.model_fields if (v := getattr(self, k)) is not None)
+        args = ",".join([f"{k}={v}" for k, v in model_dump])
+        return f"{self.type_.value}({args})"
+
+    @property
+    def id(self) -> str:
+        return self._as_str()
+
+    def _as_str(self, **defaults: Any) -> str:
+        # We have overwritten the serialization to str, so we need to do it manually
+        model_dump = {
+            field.alias or field_name: v.dump(**defaults) if isinstance(v, ConceptualEntity) else v
+            for field_name, field in self.model_fields.items()
+            if (v := getattr(self, field_name)) is not None and field_name not in {"prefix", "suffix"}
+        }
```

Comment:
I think we can manage without this one. It would be good to avoid having logic in the entity classes.